ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano.hs

```diff
@@ -154,6 +154,7 @@ data Protocol (m :: * -> *) blk p where
     -> Update.ProtocolVersion
     -> Update.SoftwareVersion
     -> Maybe PBftLeaderCredentials
+    -> HardFork.SafeBeforeEpoch
```

Comment:
(After a discussion with @edsko:) this is going to be visible in `cardano-node`. So it is good to make it easy to give the right lower bound here so that when the real hard fork will happen (e.g., epoch 182 or whatever) the "node people" can fill in the right value (if they don't, that's fine too). But let's make it a bit simpler: make the argument a `Maybe EpochNo` where `Just` = `LowerBound` and `Nothing` = `NoLowerBound` (so no `UnsafeUnbounded`, which makes sense, given its name :slightly_smiling_face:). Then copy that comment from the 180 so the "node people" know to pick `Just 180`.

Reply:
Done!